fix(core): static-query migration should gracefully exit if AOT compiler throws

devversion opened the pull request:
* See individual commits (except first one)

**Note**: Blocked on #30254

devversion committed:
fix(core): static-query migration fails with default parameter values

Currently when someone has a call expression within the `ngOnInit` call
and we try to peek into that function with respect to the current function
context, the schematic errors because a call expression argument is
undefined. This is valid because the target function declaration defines
that parameter with a default value. In order to fix this, we need to
respect parameter default values.

devversion committed:
fix(core): migrations not always migrating all files

In an Angular CLI project scenario where projects only reference
top-level source-files through the `tsconfig` `files` option, we currently
do not migrate referenced source-files. This can be fixed checking all
referenced source-files which aren't coming from an external library.

This is similar to how `tslint` determines project source-files.

CaerusKaru reviewed (approved):
LGTM

kara commented on packages/core/schematics/test/static_queries_migration_template_spec.ts:
```diff
@@ -457,17 +461,55 @@ describe('static-queries migration with template strategy', () => {
+
+      // Enable flat-module bundling in order to simulate a common AOT compiler
+      // failure that can happen in CLI projects that use flat-module bundling
+      // e.g. with ng-packager. https://github.com/angular/angular/issues/20931
```
```suggestion
      // e.g. with ng-packagr. https://github.com/angular/angular/issues/20931
```

devversion replied:
Done.

kara reviewed (approved):
LGTM, found a typo though

kara commented:
@devversion Also could you rename the commit to `fix(core)`? Refactors technically don't change behavior.

devversion committed:
fix(core): static-query migration should gracefully exit if AOT compiler throws

The static-query template strategy leverages the AOT compiler
in order to determine the query timing. Unfortunately the AOT
compiler has open bugs that can cause unexpected failures which
make the template strategy unusable in rare cases. These rare
exceptions need to be handled gracefully in order to avoid confusion
and to provide a more smooth migration.

Additionally migration strategy setup failures are now reported with
stack traces as the `ng update` command does not print stack traces.
This makes it easier to reproduce and report migration issues.

devversion commented:
@kara Done. I wasn't too sure if we want to list these changes in the changelog.

alxhub commented:
[Presubmit](https://test.corp.google.com/ui#id=OCL:247237439:BASE:247237476:1557336102801:265e8b89)
